component/file_cache/lfu_policy.go

```diff
@@ -157,8 +157,10 @@ func (l *lfuPolicy) clearItemFromCache(path string) {
 	if err == nil {
 		// File was deleted so try clearing its parent directory
 		dirPath := filepath.Dir(path)
-		if dirPath != l.tmpPath {
-			os.Remove(dirPath)
+		for ; dirPath != l.tmpPath; dirPath = filepath.Dir(dirPath) {
```

Comment:
See https://github.com/Azure/azure-storage-fuse/pull/765

Reply:
This fix won't work either. We will have to work on something more complex than this to prevent synchronization issues.

Reply:
This fix won't work either. We will have to work on something more complex than this to prevent synchronization issues.

Reply:
Ya got your point from your PR. Will remove this deletion of parent directory logic completly.